content/BPIP-9.md

```diff
@@ -0,0 +1,250 @@
+
+```diff solidity
+    /**
+     * @notice Commits to a static offer (first step of an exchange).
+     *
+-    * Emits a BuyerCommitted event if successful.
++    * Emits a BuyerCommitted or SellerCommitted event if successful.
+     * Issues a voucher to the buyer address.
+     *
+     * Reverts if:
+     * - The exchanges region of protocol is paused
+     * - The buyers region of protocol is paused
++    * - The sellers region of protocol is paused
+     * - OfferId is invalid
+     * - Offer has been voided
+     * - Offer has expired
+     * - Offer is not yet available for commits
+     * - Offer's quantity available is zero
+-    * - Buyer address is zero
+-    * - Buyer account is inactive
++    * - Committer address is zero
++    * - Committer account is inactive
+     * - Buyer is token-gated (conditional commit requirements not met or already used)
+-    * - Offer price is in native token and caller does not send enough
+-    * - Offer price is in some ERC20 token and caller also sends native currency
++    * - Offer exchange token is in native token and caller does not send enough
++    * - Offer exchange token is in some ERC20 token and caller also sends native currency
+     * - Contract at token address does not support ERC20 function transferFrom
+     * - Calling transferFrom on token fails for some reason (e.g. protocol is not approved to transfer)
+     * - Received ERC20 token amount differs from the expected value
+-    * - Seller has less funds available than sellerDeposit
++    * - Seller has less funds available than sellerDeposit if offer was created by the seller
++    * - Buyer has less funds available than price if offer was created by the buyer
+     *
+-    * @param _buyer - the buyer's address (caller can commit on behalf of a buyer)
++    * @param _committer - the seller's or the buyer's address. The caller can commit on behalf of a buyer or a seller.
+     * @param _offerId - the id of the offer to commit to
+     */
+-   function commitToOffer(address payable _buyer, uint256 _offerId) external payable;
++   function commitToOffer(address payable _committer, uint256 _offerId) external payable;
+```
+
+#### IBosonPriceDiscoveryHandler
+```diff solidity
+    /**
+     * @notice Commits to a price discovery offer (first step of an exchange).
+     *
+-    * Emits a BuyerCommitted event if successful.
++    * Emits a BuyerCommitted or SellerCommitted event if successful.
+     * Issues a voucher to the buyer address.
+     *
+     * Reverts if:
+     * - Offer price type is not price discovery. See BosonTypes.PriceType
+     * - Price discovery contract address is zero
+     * - Price discovery calldata is empty
+     * - Exchange exists already
+     * - Offer has been voided
+     * - Offer has expired
+     * - Offer is not yet available for commits
++    * - Committer address is zero
++    * - Committer account is inactive
```

Comment:
What is "inactive"?

Reply:
It means that `Buyer.active` is set to false for whatever reason. We use the same wording as with other methods.  
Maybe a better word is `deactivated` (although there is currently no way to deactivate the account).